datacapture/src/main/java/com/google/android/fhir/datacapture/views/factories/DropDownViewHolderFactory.kt

```diff
@@ -97,32 +102,69 @@ internal object DropDownViewHolderFactory :
               null,
             )
           }
+
         autoCompleteTextView.setAdapter(adapter)
         autoCompleteTextView.onItemClickListener =
           AdapterView.OnItemClickListener { _, _, position, _ ->
-            val selectedItem = adapter.getItem(position)
-            autoCompleteTextView.setText(selectedItem?.answerOptionString, false)
-            autoCompleteTextView.setCompoundDrawablesRelative(
-              adapter.getItem(position)?.answerOptionImage,
-              null,
-              null,
-              null,
-            )
-            val selectedAnswer =
-              questionnaireViewItem.enabledAnswerOptions
-                .firstOrNull { it.value.identifierString(context) == selectedItem?.answerId }
-                ?.value
-
-            if (selectedAnswer == null) {
-              questionnaireViewItem.clearAnswer()
-            } else {
-              questionnaireViewItem.setAnswer(
-                QuestionnaireResponse.QuestionnaireResponseItemAnswerComponent()
-                  .setValue(selectedAnswer),
+            if (isDropdownEditable) {
+              val selectedItem = adapter.getItem(position)
+              autoCompleteTextView.setText(selectedItem?.answerOptionString, false)
+              autoCompleteTextView.setCompoundDrawablesRelative(
+                adapter.getItem(position)?.answerOptionImage,
+                null,
+                null,
+                null,
               )
+
+              isDropdownEditable = false
+              val selectedAnswer =
+                questionnaireViewItem.enabledAnswerOptions
+                  .firstOrNull { it.value.identifierString(context) == selectedItem?.answerId }
+                  ?.value
+
+              if (selectedAnswer == null) {
+                questionnaireViewItem.clearAnswer()
+              } else {
+                questionnaireViewItem.setAnswer(
+                  QuestionnaireResponse.QuestionnaireResponseItemAnswerComponent()
+                    .setValue(selectedAnswer),
+                )
+              }
             }
           }
 
+        autoCompleteTextView.addTextChangedListener(
+          object : TextWatcher {
+            override fun beforeTextChanged(
+              charSequence: CharSequence?,
+              start: Int,
+              count: Int,
+              after: Int,
+            ) {}
+
+            override fun onTextChanged(
+              charSequence: CharSequence?,
+              start: Int,
+              before: Int,
+              count: Int,
+            ) {}
+
+            override fun afterTextChanged(editable: Editable?) {
+              if (editable.isNullOrBlank()) {
+                Handler(Looper.getMainLooper())
+                  .postDelayed(
+                    {
+                      if (autoCompleteTextView.isPopupShowing.not()) {
+                        autoCompleteTextView.showDropDown()
+                      }
+                    },
+                    100,
```

Comment:
```suggestion
                autoCompleteTextView.post(
                    {
                      if (autoCompleteTextView.isPopupShowing.not()) {
                        autoCompleteTextView.showDropDown()
                      }
                    }
```

Can you check if this helps. Anyways, we should just do a `post` / `postDelayed` on the `autoCompleteTextView`

Reply:
I've refactored that to
https://github.com/google/android-fhir/pull/2393/commits/12fb10b5ba7c89681d7d812e3a1d4d9709a7dd3d